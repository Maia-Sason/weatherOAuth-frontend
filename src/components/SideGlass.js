import {
  makeStyles,
  Paper,
  Box,
  Grid,
  Divider,
  Typography,
  Fade,
  Grow,
} from "@material-ui/core";
import { useState } from "react";
import AsideWeather from "./AsideWeather";
import AsideTemp from "./AsideTemp";
import DayWeatherCard from "./DayWeatherCard";

const useStyles = makeStyles((theme) => ({
  sidePaper: {
    height: "100vh",
    backgroundColor: "rgba(100, 100, 100, .15)",
    overflowY: "scroll",
    backdropFilter: "blur(5px)",
    color: "white",
    padding: "20px",
  },
  headerC: {
    fontSize: "4em",
    textAlign: "center",
    letterSpacing: "3px",
    fontWeight: "100",
  },
  lowHigh: {
    display: "flex",
    justify: "space-between",
  },
}));

const SideGlass = ({ days, load, rain, wind, high, low, feels }) => {
  const [anim, setLoad] = useState(false);

  useState(() => {
    setTimeout(function () {
      setLoad(true);
    }, 2000);
  }, [load]);

  console.log(days);
  const weatherCardList = days.slice(1).map((item, k) => {
    const num = 1000 + 1000 * k;
    return (
      <Grow in={anim} {...(anim ? { timeout: num } : {})}>
        <Grid item xs={6} key={k}>
          <DayWeatherCard
            day={item.day * 1000}
            icon={item.icon}
            temp={item.temp.toFixed(0)}
          />
        </Grid>
      </Grow>
    );
  });

  const classes = useStyles();

  return (
    <>
      <Paper className={classes.sidePaper} square>
        <Box>
          <Typography className={classes.headerC}>Today,</Typography>
          <Grid container justify={"center"}>
            <Grid item xs={8}>
              <AsideWeather
                feels={feels.toFixed(0)}
                rain={rain}
                wind={wind.toFixed(0)}
              />
            </Grid>
          </Grid>
          <Grid
            container
            direction="row"
            justify="space-between"
            alignItems="flex-start"
          >
            <Grid item xs={4}>
              <AsideTemp temp={high.toFixed(0)} text={"High"}></AsideTemp>
            </Grid>
            <Grid item xs={4}>
              <AsideTemp temp={low.toFixed(0)} text={"Low"}></AsideTemp>
            </Grid>
          </Grid>
        </Box>
        <Divider></Divider>
        <Box>
          <Typography className={classes.headerC}>This Week,</Typography>
          <Grid container style={{ padding: "20px" }} justify="center">
            {weatherCardList}
          </Grid>
        </Box>
      </Paper>
    </>
  );
};

export default SideGlass;
