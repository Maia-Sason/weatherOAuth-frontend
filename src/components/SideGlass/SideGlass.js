import {
  makeStyles,
  Paper,
  Box,
  Grid,
  Divider,
  Typography,
  Fade,
  Tabs,
  Tab,
  Grow,
} from "@material-ui/core";
import { useState } from "react";
import AsideWeather from "../AsideWeather/AsideWeather";
import AsideTemp from "../AsideTemp/AsideTemp";
import DayWeatherCard from "../DayWeatherCard/DayWeatherCard";
import { TabPanel, TabContext } from "@material-ui/lab";

const useStyles = makeStyles((theme) => ({
  sidePaper: {
    height: "100vh",
    backgroundColor: "rgba(100, 100, 100, .15)",
    overflowY: "scroll",
    backdropFilter: "blur(5px)",
    color: "white",
    padding: "20px",
  },
  headerC: {
    fontSize: "4em",
    textAlign: "center",
    letterSpacing: "3px",
    fontWeight: "100",
  },
  lowHigh: {
    display: "flex",
    justify: "space-between",
  },
}));

const SideGlass = ({ days, load, rain, wind, high, low, feels }) => {
  const [anim, setLoad] = useState(false);
  const [value, setValue] = useState(0);

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  useState(() => {
    setTimeout(function () {
      setLoad(true);
    }, 2000);
  }, [load]);

  // function a11yProps(index) {
  //   return {
  //     id: `simple-tab-${index}`,
  //     "aria-controls": `simple-tabpanel-${index}`,
  //   };
  // }

  console.log(days);
  const weatherCardList = days.slice(1).map((item, k) => {
    const num = 1000 + 1000 * k;
    return (
      <Grow key={k} in={anim} {...(anim ? { timeout: num } : {})}>
        <Grid item xs={6} key={k}>
          <DayWeatherCard
            key={k}
            day={item.day * 1000}
            icon={item.icon}
            temp={item.temp.toFixed(0)}
          />
        </Grid>
      </Grow>
    );
  });

  const classes = useStyles();

  return (
    <>
      <TabContext value={value}>
        <Paper className={classes.sidePaper} square>
          <TabPanel value={value} index={0}>
            <Box>
              <Typography className={classes.headerC}>Today,</Typography>
              <Grid container justify={"center"}>
                <Grid item xs={8}>
                  <AsideWeather
                    feels={feels.toFixed(0)}
                    rain={rain}
                    wind={wind.toFixed(0)}
                  />
                </Grid>
              </Grid>
              <Grid
                container
                direction="row"
                justify="space-between"
                alignItems="flex-start"
              >
                <Grid item xs={4}>
                  <AsideTemp temp={high.toFixed(0)} text={"High"}></AsideTemp>
                </Grid>
                <Grid item xs={4}>
                  <AsideTemp temp={low.toFixed(0)} text={"Low"}></AsideTemp>
                </Grid>
              </Grid>
            </Box>
            <Divider></Divider>
            <Box>
              <Typography className={classes.headerC}>This Week,</Typography>
              <Grid container style={{ padding: "20px" }} justify="center">
                {weatherCardList}
              </Grid>
            </Box>
          </TabPanel>
          <TabPanel value={value} index={1}>
            <Typography className={classes.headerC}>Second Tab</Typography>
          </TabPanel>
          <Paper>
            <Tabs
              value={value}
              onChange={handleChange}
              variant="fullWidth"
              indicatorColor="primary"
              textColor="primary"
            >
              <Tab label="Current" value={0}></Tab>
              <Tab label="All" value={1}></Tab>
            </Tabs>
          </Paper>
        </Paper>
      </TabContext>
    </>
  );
};

export default SideGlass;
