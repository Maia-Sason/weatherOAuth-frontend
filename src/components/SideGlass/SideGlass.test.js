import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import SideGlass from "./SideGlass";
import { ExpansionPanelActions } from "@material-ui/core";

test("<SideGlass /> matches snapshot", () => {
  const days = [
    {
      temp: 75.11,
      precipitation: 1,
      icon: "10d",
      day: 1625270400,
    },
    {
      temp: 59.34,
      precipitation: 0.92,
      icon: "10d",
      day: 1625356800,
    },
    {
      temp: 69.53,
      precipitation: 0.94,
      icon: "10d",
      day: 1625443200,
    },
    {
      temp: 77.14,
      precipitation: 0,
      icon: "04d",
      day: 1625529600,
    },
    {
      temp: 85.01,
      precipitation: 0,
      icon: "01d",
      day: 1625616000,
    },
  ];

  const component = render(
    <SideGlass
      days={days}
      feels={99.3}
      rain={0.3}
      wind={12.6}
      high={100.33}
      low={67.89}
    />
  );
  expect(component.container).toMatchSnapshot();
});

it("Renders SideGlass without crashing", () => {
  const div = document.createElement("div");

  const days = [
    {
      temp: 75.11,
      precipitation: 1,
      icon: "10d",
      day: 1625270400,
    },
    {
      temp: 59.34,
      precipitation: 0.92,
      icon: "10d",
      day: 1625356800,
    },
    {
      temp: 69.53,
      precipitation: 0.94,
      icon: "10d",
      day: 1625443200,
    },
    {
      temp: 77.14,
      precipitation: 0,
      icon: "04d",
      day: 1625529600,
    },
    {
      temp: 85.01,
      precipitation: 0,
      icon: "01d",
      day: 1625616000,
    },
  ];

  ReactDOM.render(
    <SideGlass
      days={days}
      feels={99.3}
      rain={0.3}
      wind={12.6}
      high={100.33}
      low={67.89}
    />,
    div
  );
});
