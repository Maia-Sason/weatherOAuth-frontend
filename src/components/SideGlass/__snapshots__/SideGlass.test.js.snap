// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideGlass /> matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-sidePaper-1 MuiPaper-elevation1"
  >
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <p
        class="MuiTypography-root makeStyles-headerC-2 MuiTypography-body1"
      >
        Today,
      </p>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
        >
          <div
            class="MuiPaper-root makeStyles-container-5 MuiPaper-elevation10 MuiPaper-rounded"
          >
            <div
              class="MuiBox-root MuiBox-root-12 makeStyles-headerBox-6"
            >
              <p
                class="MuiTypography-root makeStyles-header-7 MuiTypography-body1"
              >
                Feels Like:
              </p>
              <p
                class="MuiTypography-root makeStyles-header-7 MuiTypography-body1"
              >
                99
                °
              </p>
            </div>
            <hr
              class="MuiDivider-root"
            />
            <div
              class="MuiBox-root MuiBox-root-13 makeStyles-precipitation-10"
            >
              <svg
                class="makeStyles-rainDrop-9"
              >
                drop-percentage.svg
              </svg>
              <p
                class="MuiTypography-root makeStyles-headerB-8 MuiTypography-body1"
              >
                30
                %
              </p>
            </div>
            <hr
              class="MuiDivider-root"
            />
            <div
              class="MuiBox-root MuiBox-root-14 makeStyles-wind-11"
            >
              <svg
                class="makeStyles-rainDrop-9"
              >
                wind-2.svg
              </svg>
              <p
                class="MuiTypography-root makeStyles-headerB-8 MuiTypography-body1"
              >
                13
                mph
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            class="MuiBox-root MuiBox-root-16"
          >
            <p
              class="MuiTypography-root makeStyles-text-15 MuiTypography-body1"
            >
              High
               
              100
              °
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <div
            class="MuiBox-root MuiBox-root-17"
          >
            <p
              class="MuiTypography-root makeStyles-text-15 MuiTypography-body1"
            >
              Low
               
              68
              °
            </p>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiBox-root MuiBox-root-18"
    >
      <p
        class="MuiTypography-root makeStyles-headerC-2 MuiTypography-body1"
      >
        This Week,
      </p>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
        style="padding: 20px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        >
          <div
            class="MuiPaper-root makeStyles-paper-19 MuiPaper-elevation10 MuiPaper-rounded"
          >
            <div
              class="MuiBox-root MuiBox-root-24 makeStyles-box-20"
            >
              <p
                class="MuiTypography-root makeStyles-text-22 MuiTypography-body1"
              >
                59
                °
              </p>
              <div
                class="MuiBox-root MuiBox-root-25 makeStyles-icon-21"
              >
                <svg>
                  rain-day.svg
                </svg>
              </div>
            </div>
            <p
              class="MuiTypography-root makeStyles-day-23 MuiTypography-body1"
            >
              Saturday
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        >
          <div
            class="MuiPaper-root makeStyles-paper-19 MuiPaper-elevation10 MuiPaper-rounded"
          >
            <div
              class="MuiBox-root MuiBox-root-26 makeStyles-box-20"
            >
              <p
                class="MuiTypography-root makeStyles-text-22 MuiTypography-body1"
              >
                70
                °
              </p>
              <div
                class="MuiBox-root MuiBox-root-27 makeStyles-icon-21"
              >
                <svg>
                  rain-day.svg
                </svg>
              </div>
            </div>
            <p
              class="MuiTypography-root makeStyles-day-23 MuiTypography-body1"
            >
              Sunday
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        >
          <div
            class="MuiPaper-root makeStyles-paper-19 MuiPaper-elevation10 MuiPaper-rounded"
          >
            <div
              class="MuiBox-root MuiBox-root-28 makeStyles-box-20"
            >
              <p
                class="MuiTypography-root makeStyles-text-22 MuiTypography-body1"
              >
                77
                °
              </p>
              <div
                class="MuiBox-root MuiBox-root-29 makeStyles-icon-21"
              >
                <svg>
                  mostly-cloudy-2.svg
                </svg>
              </div>
            </div>
            <p
              class="MuiTypography-root makeStyles-day-23 MuiTypography-body1"
            >
              Monday
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        >
          <div
            class="MuiPaper-root makeStyles-paper-19 MuiPaper-elevation10 MuiPaper-rounded"
          >
            <div
              class="MuiBox-root MuiBox-root-30 makeStyles-box-20"
            >
              <p
                class="MuiTypography-root makeStyles-text-22 MuiTypography-body1"
              >
                85
                °
              </p>
              <div
                class="MuiBox-root MuiBox-root-31 makeStyles-icon-21"
              >
                <svg>
                  sun-1.svg
                </svg>
              </div>
            </div>
            <p
              class="MuiTypography-root makeStyles-day-23 MuiTypography-body1"
            >
              Tuesday
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
